### PR TITLE
adding rename mappings

### PR DESCRIPTION
**Pull request by slemasne**
# Pull Request Checklist

- [x] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: lusidtools/pandas_utils/lusid_pandas.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ def lusid_response_to_data_frame(lusid_response, rename_properties: bool = False
     :param lusid_response: a response from the LUSID APIs (e.g. VersionedResourceListOfPortfolioHolding)
     :param bool rename_properties: this parameter formats the returned DataFrame. Specifically, the formatter does two things;
     (1) removes any metadata columns for properties and SHKs, and (2) simplifies the naming of column headers for properties and SHKs.
+    :param dict rename_mapping: a dictionary which is used to map old column headers to new ones. The dictionary key is old header
+    while the dictionary value in new header.
```
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: add a comment saying what happens if the given property name doesn't exist

File: lusidtools/pandas_utils/lusid_pandas.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from flatten_json import flatten
 
 
-def lusid_response_to_data_frame(lusid_response, rename_properties: bool = False):
+def lusid_response_to_data_frame(lusid_response, rename_properties: bool = False, rename_mapping: dict = None):
```
Comment (rizwansaeed) [inline lusidtools/pandas_utils/lusid_pandas.py]: maybe rename to something `column_names`

File: tests/unit/pandas_utils/test_lusid_pandas.py [hunk 3]
```diff
@@ -157,3 +157,68 @@ def test_response_to_df_format(self):
...
+        )
+
+        holdings_df = lusid_response_to_data_frame(
+            test_holdings_response,
+            rename_properties=True,
+            rename_mapping=rename_mapping,
+        )
+
+        df_columns = holdings_df.columns
+
```
Comment (rizwansaeed) [inline tests/unit/pandas_utils/test_lusid_pandas.py]: add asserts showing the mapped column names don't exist

File: tests/unit/pandas_utils/test_lusid_pandas.py [hunk 4]
```diff
@@ -157,3 +157,68 @@ def test_response_to_df_format(self):
...
+        self.assertIsInstance(holdings_df, pd.DataFrame)
+        self.assertEqual(holdings_df.loc[0]["Instrument ID"], "LUID_123")
+        self.assertIn("PortionClass(MultiAssetScope-SubHoldingKeys)", df_columns)
+        self.assertTrue(
```
Comment (rizwansaeed) [inline tests/unit/pandas_utils/test_lusid_pandas.py]: can this be made `self.assertEquals` as you get better error messages when it fails (as opposed to 'Expected True got False)

File: tests/unit/pandas_utils/test_lusid_pandas.py [hunk 5]
```diff
@@ -157,3 +157,68 @@ def test_response_to_df_format(self):
...
+            test_holdings_response,
+            rename_properties=False,
+            rename_mapping=rename_mapping,
+        )
+
+        df_columns = holdings_df.columns
+
+        print(df_columns)
+
+        self.assertIsInstance(holdings_df, pd.DataFrame)
```
Comment (rizwansaeed) [inline tests/unit/pandas_utils/test_lusid_pandas.py]: as above

File: tests/unit/pandas_utils/test_lusid_pandas.py [hunk 6]
```diff
@@ -157,3 +157,68 @@ def test_response_to_df_format(self):
...
+        self.assertEqual(holdings_df.loc[0]["Instrument ID"], "LUID_123")
+        self.assertTrue(
+            set(
+                ["Holding Type", "Units", "Settled Units", "Quantity", "Cost Currency"]
+            ).issubset(set(df_columns))
+        )
```
Comment (rizwansaeed) [inline tests/unit/pandas_utils/test_lusid_pandas.py]: as discussed, add a test for when `rename_properties ` and `rename_properties` refer to the same column

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 1 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Sure, done.

File: lusidtools/pandas_utils/lusid_pandas.py (hunk 2 above)
Comment (slemasne) [reply lusidtools/pandas_utils/lusid_pandas.py]: Sure, renamed to "column_name_mapping"

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 6 above)
Comment (slemasne) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: Sure, thats done.
Added test:
test_response_to_df_rename_property

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 5 above)
Comment (slemasne) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: Thats done.
Updated to `self.assertEquals`

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 4 above)
Comment (slemasne) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: Thats done.
Updated to `self.assertEquals`

File: tests/unit/pandas_utils/test_lusid_pandas.py (hunk 3 above)
Comment (slemasne) [reply tests/unit/pandas_utils/test_lusid_pandas.py]: Sure, done via `self.assertNotIn`

**Pull request merged**